### PR TITLE
*: add the DM solution for sharding data

### PR DESCRIPTION
**Pull request by lilin90**
@GregoryIan @csuzhangxc @CaitinChen PTAL

This PR involves a link that points to the document in https://github.com/pingcap/docs/pull/755, so we should merge this PR after https://github.com/pingcap/docs/pull/755 is merged, to avoid the broken link issue.

Via: 
- https://github.com/pingcap/tidb-tools/blob/docs/docs/dm/zh_CN/shard-table/merge-solution.md
- https://github.com/pingcap/tidb-tools/blob/docs/docs/dm/zh_CN/shard-table/restrictions.md

### CODE REVIEW

File: tools/dm-sharding-solution.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+To complete the sharding data synchronization task, Data Migration uses the global DDL lock in the following two layers of DDL synchronization:
+
+- The first layer: It is the DDL synchronization within the DM-worker. The synchronization related information is saved by DM-worker. Only after the synchronization is finished at this layer, the second layer of synchronization is carried out.
+- The second layer: The synchronization is executed by coordinating the DM-workers using the global DDL lock. The synchronization related information is saved by both DM-master and DM-worker.
+
+The processing details are as follows:
+
+1. DM-worker creates a sharding group for all upstream tables merged into the same target table (first layer synchronization; members are the upstream tables)
+
+2. DM-master creates a sharding group for all DM-workers that perform the target table merging task (the second layer synchronization; the members are all dm-workers)
```
Comment (CaitinChen) [inline tools/dm-sharding-solution.md]:
```suggestion
2. DM-master creates a sharding group for all DM-workers that perform the target table merging task (the second layer synchronization; the members are all DM-workers)
```

File: tools/dm-sharding-solution.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    
+    - If some of the sharding DDL statements have been executed when you start to synchronize the synchronization points, the sharding DDL operation can never be synchronized successfully.
+
+- If you need to change `router-rules`, you need to wait for all sharding DDL synchronization to complete.
+
+    - While in the process of sharding DDL synchronization, it reports an error if you use dmctl to change `router-rules`.
+
+- If you need to `CREATE` a new table to an existing sharding group, you need to keep the table schema the same as the newly edited table schema.
+
+    - For example, the original table_a, table_b initially has two columns (a, b), and after the sharding DDL has three columns (a, b, c), the table of the new CREATE after the synchronization is completed should have three columns (a, b, c).
```
Comment (CaitinChen) [inline tools/dm-sharding-solution.md]:
```suggestion
    - For example, the original table_a, table_b initially has two columns (a, b), and after the sharding DDL has three columns (a, b, c), the table of the new `CREATE` after the synchronization is completed should have three columns (a, b, c).
```

File: tools/dm-sharding-solution.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+- The sharding group synchronization task supports `RENAME TABLE`, but with the following limits:
+
+   - A table can only be renamed to a new name that is not used by any other table.
+   - A single `RENAME TABLE` statement can only involve a single `RENAME` operation. (Online DDL is supported in another solution)
+
+- The synchronization point where each table starts the task should be before or after all sharding DDL operations that need to be synchronized
+    
+    - If some of the sharding DDL statements have been executed when you start to synchronize the synchronization points, the sharding DDL operation can never be synchronized successfully.
+
+- If you need to change `router-rules`, you need to wait for all sharding DDL synchronization to complete.
```
Comment (CaitinChen) [inline tools/dm-sharding-solution.md]:
```suggestion
- If you need to change `router-rules`, you need to wait for synchronization of all sharding DDL operations to complete.
```

**Comment by lilin90** [comment]
@CaitinChen Thanks! Comments addressed. PTAL again

**Comment by csuzhangxc** [comment]
LGTM

**Comment by lilin90** [comment]
@GregoryIan @morgo PTAL 
We'll merge this PR after #755 is merged.

**Pull request merged**